src/caffe/blob.cpp

```diff
@@ -205,6 +205,142 @@ Dtype Blob<Dtype>::asum_diff() const {
   return 0;
 }
 
+template <> unsigned int Blob<unsigned int>::sumsq_data() const {
+  NOT_IMPLEMENTED;
+  return 0;
+}
+
+template <> int Blob<int>::sumsq_data() const {
+  NOT_IMPLEMENTED;
+  return 0;
+}
+
+template <typename Dtype>
+Dtype Blob<Dtype>::sumsq_data() const {
```

Comment:
I don't like writing or reading these duplicate methods for `data` and `diff` -- should these be in `SyncedMemory`, or perhaps a new abstraction layer between `SyncedMemory` and `Blob`?